urlchecker/core/urlproc.py

```diff
@@ -90,28 +90,61 @@ def get_user_agent():
     return random.choice(agents)
 
 
+def find_urls(file_names, exclude_patterns=None, exclude_urls=None):
+    """
+    Given a list of file names, return the file name and a list of urls to parse
+    (intended to be passed on to the UrlCheckResultResult class) that removes
+    any redundant URLs.
+    """
+    # keep a master list of urls we have seen - no repeats allowed!
+    seen = set()
+
+    for file_name in file_names:
+        checker = UrlCheckResult(
+            file_name=file_name,
+            exclude_patterns=exclude_patterns,
+            exclude_urls=exclude_urls,
+        )
+
+        # Ensure we don't have repeats and updated seen list
+        checker.urls = [u for u in checker.urls if u not in seen]
```

Comment:
This is not totally correct imo, say you have a file with the same link twice; now if that link came up in `seen` before then none of the links will be added. However, if the link did not come up in `seen`, then both links will be written to the list. Maybe we should add a `list(set([u for u in checker.urls if u not in seen]))` here ?

Reply:
The function that derives the checker urls the first time around is already returning a unique set. I think this is right.

Reply:
If the returned list is unique then this is fine ;)